tests/unittests/algo/test_evolution_es.py

```diff
@@ -520,3 +520,29 @@ def assert_callbacks(
         for j in range(0, rung_id + 1):
             for bracket in brackets:
                 assert len(bracket.rungs[j]["results"]) > 0, (bracket, j)
+
+    @pytest.mark.parametrize("seed", [123, 456])
+    def test_state_dict(self, seed: int):
+        """Verify that resetting state makes sampling deterministic"""
+        algo = self.create_algo(seed=seed)
+
+        state = algo.state_dict
+        a = algo.suggest(1)[0]
+
+        # NOTE: This is not necessarily true for all algorithms. For instance, if the algo doesn't
+        # have any RNG (e.g. GridSearch), this will fail.
+        new_algo = self.create_algo()
+        new_state = new_algo.state_dict
+        b = new_algo.suggest(1)[0]
+        if _are_equal(new_state, state):
+            # If the state is the same, the trials should be the same.
+            assert a == b
+        else:
+            # If the state is different, the trials should be different.
+            assert a != b
+
+        new_algo.set_state(state)
+        c = new_algo.suggest(1)[0]
+        # TODO: For EvolutionES, the params are identical, but the ids are different.
```

Comment:
That's concerning. Did you found out why? Please let me know if you would need help to investigate this.

Reply:
(Also this should probably be another issue/PR, not work for this one)

Reply:
Created a new issue for this: https://github.com/Epistimio/orion/issues/890

I did not find out why, but I know that the trial's parents has a different ID as well (IDK if that's useful).

Reply:
Thanks!